cvat/apps/auto_annotation/model_loader.py

```diff
@@ -47,22 +52,35 @@ def __init__(self, model, weights):
 
     def infer(self, image):
         _, _, h, w = self._input_layout
-        in_frame = image if image.shape[:-1] == (h, w) else cv2.resize(image, (w, h))
+        scale = min(h / image.shape[0], w / image.shape[1])
+        in_frame = image if image.shape[:-1] == (h, w) else cv2.resize(image, None, fx=scale, fy=scale)
+
+        in_frame_size = in_frame.shape[:2]
+        in_frame = np.pad(in_frame, ((0, h - in_frame_size[0]),
+                                     (0, w - in_frame_size[1]),
+                                     (0, 0)),
+                          mode='constant', constant_values=0)
+
         in_frame = in_frame.transpose((2, 0, 1))  # Change data layout from HWC to CHW
         inputs = {self._input_blob_name: in_frame}
         if self._require_image_info:
-            info = np.zeros([1, 3])
-            info[0, 0] = h
-            info[0, 1] = w
-            # frame number
-            info[0, 2] = 1
-            inputs['image_info'] = info
+            info = np.asarray([[in_frame_size[0],
+                                in_frame_size[1],
+                                scale]],
+                              dtype=np.float32)
+
+            inputs[self._input_info_name] = info
 
         results = self._net.infer(inputs)
+
+        if scale != 1:
+            results['blob_height'] = h
+            results['blob_width'] = w
```

Comment:
Can probably hardcode the blob sizes in the interp file to avoid breaking this interface.